libs/ui-lib/lib/ocm/components/AddHosts/day2Wizard/Day2ClusterDetails.tsx

```diff
@@ -23,24 +26,64 @@ import Day2WizardFooter from './Day2WizardFooter';
 import Day2HostStaticIpConfigurations from './Day2StaticIpHostConfigurations';
 import { mapClusterCpuArchToInfraEnvCpuArch } from '../../../services/CpuArchitectureService';
 import CpuArchitectureDropdown from '../../clusterConfiguration/CpuArchitectureDropdown';
-import { useOpenshiftVersions } from '../../../hooks';
+import { useOpenshiftVersions, usePullSecret } from '../../../hooks';
+
+const createNewInfraEnvWithCpuArchitecture = (
+  pullSecret: string,
+  cpuArchitecture: ClusterCpuArchitecture,
+  id: string,
+  name?: string,
+  openshiftVersion?: string,
+) => {
+  //If infraEnv don't exist, create a new one
+  return InfraEnvsService.create({
+    name: `${name || ''}_infra-env-${cpuArchitecture}`,
+    pullSecret,
+    clusterId: id,
+    openshiftVersion: openshiftVersion,
+    cpuArchitecture: cpuArchitecture,
+  });
+};
 
 const getDay2ClusterDetailInitialValues = async (
   clusterId: Cluster['id'],
   day1CpuArchitecture: CpuArchitecture,
+  pullSecret?: string,
+  name?: string,
+  openshiftVersion?: string,
 ) => {
   try {
     const infraEnv = await InfraEnvsService.getInfraEnv(clusterId, day1CpuArchitecture);
-
-    return {
-      cpuArchitecture: day1CpuArchitecture,
-      hostsNetworkConfigurationType: infraEnv.staticNetworkConfig
-        ? HostsNetworkConfigurationType.STATIC
-        : HostsNetworkConfigurationType.DHCP,
-    };
+    if (infraEnv) {
+      return {
+        cpuArchitecture: day1CpuArchitecture,
+        hostsNetworkConfigurationType: infraEnv.staticNetworkConfig
+          ? HostsNetworkConfigurationType.STATIC
+          : HostsNetworkConfigurationType.DHCP,
+      };
+    } else {
+      //Create a new InfraEnv because is not found
+      if (pullSecret) {
+        const infraEnv = await createNewInfraEnvWithCpuArchitecture(
+          pullSecret,
+          day1CpuArchitecture as ClusterCpuArchitecture,
+          clusterId,
+          name,
+          openshiftVersion,
+        );
+        return {
+          cpuArchitecture: day1CpuArchitecture,
+          hostsNetworkConfigurationType: infraEnv.staticNetworkConfig
+            ? HostsNetworkConfigurationType.STATIC
+            : HostsNetworkConfigurationType.DHCP,
+        };
+      } else {
+        throw Error('Could not get infraEnv for this cpu architecture');
```

Comment:
I don't think this is a good idea...

Reply:
I've added this, to show the "Error state":
```
if (!initialValues) {
    return <LoadingState />;
  } else if (initialValues instanceof Error) {
    return <ErrorState content="Failed to load associated data for the Day2 cluster" />;
  }
```